ledger-keyring-mv2.js

```diff
@@ -0,0 +1,195 @@
+const { BaseLedgerKeyring } = require('./base-ledger-keyring')
+
+const CONNECTION_EVENT = 'ledger-connection-change'
+
+class LedgerKeyringMv2 extends BaseLedgerKeyring {
+  init () {
+    this.iframeLoaded = false
+    this._setupIframe()
+
+    this.currentMessageId = 0
+    this.messageCallbacks = {}
+    this._setupListener()
+
+    return Promise.resolve()
+  }
+
+  destroy () {
+    window.removeEventListener('message', this._eventListener)
+
+    return Promise.resolve()
+  }
+
+  _setupIframe () {
+    this.iframe = document.createElement('iframe')
+    this.iframe.src = this.bridgeUrl
+    this.iframe.allow = `hid 'src'`
+    this.iframe.onload = async () => {
+      // If the ledger live preference was set before the iframe is loaded,
+      // set it after the iframe has loaded
+      this.iframeLoaded = true
+      if (this.delayedPromise) {
+        try {
+          const result = await this.updateTransportMethod(
+            this.delayedPromise.transportType,
+          )
+          this.delayedPromise.resolve(result)
```

Comment:
Kind of nit-picky, and potentially a breaking change, but if someone `init`'s the keyring twice, or calls `updateTransportMethod` twice quickly while the iframe is loading then you could run into situations where this `if` block runs but when `this.delayedPromise.resolve(result)` or `this.delayedPromise.reject(e)` is called the object is deleted by a separate execution context.

Ways that this could be fixed:
* Let sleeping dogs lie - as it's not a massive issue
* change `this.delayedPromise` to `this.delayedPromises: Array<Object> = []`, and then run something like:
```
for (const delayedPromise of this.delayedPromises) {
    this.updateTransportMethod(delayedPromise.transportType)
         .then((res) => delayedPromise.resolve(res))
         .catch(e => delayedPromise.reject(e))
}

this.delayedPromises = [];

// and then when the promise is added we do 
this.delayedPromises.push({
 // ...
})
```
* or, to allow for more flexibility, `this.delayedPromise` could be changed to something like `this.iframeOnloadHooks: Array<Promise>` and the whole `this.updateTransportMethod` could be contained in the array, in case someone needs to add in a different onload call in later?